plugins/woocommerce/client/blocks/assets/js/blocks/add-to-cart-with-options/grouped-product-selector/product-item-template/edit.tsx

```diff
@@ -87,32 +88,68 @@ export default function ProductItemTemplateEdit(
 	} );
 
 	const { product } = useProductDataContext();
+	const [ products, setProducts ] = useState< ProductResponseItem[] | null >(
+		null
+	);
 
-	const { products } = useSelect(
-		( select ) => {
-			const postType = 'product';
-			const { getEntityRecords } = select( coreStore );
+	useEffect( () => {
+		const fetchGroupedProducts = async (
+			productContext: ProductResponseItem[]
```

Comment:
What do you think about this function taking `groupedProductIds` as the parameter instead of `productContext`? I find it a bit confusing right now because `productContext` might contain the grouped product or the child products, and we handle it differently by looking at `productContext.length`.

If we change it, we could even rename the function to `fetchChildProducts()` or something along those lines so it's clearer it's retrieving the child products and not the parent grouped product.

Reply:
Great idea, thank you! I've renamed this function to `fetchChildProducts`, and it now takes an array of product IDs, which means they can be passed directly to the query. This looks much simpler 🎉 Let me know if I've missed anything with the latest changes.